### DISCUSSION TITLE
The "Create" form should convert to an "Update" form if a value exists in the table and auto-fill the form fields.

### DISCUSSION THREAD

**Discussion by Athik321** [Budibase/budibase#14168] [Help] [answered]
I am creating a form and I am using MySQL as the data source. In my form, there are various fields. I want to implement functionality such that when I select a value from a dropdown, if it already exists in the same table, all the form fields should auto-fill with the corresponding values from the table. When I submit the form, it should update the existing values.

**Comment by mikesealey** [comment, marked as answer]
Hey @Athik321 

I think what you're looking for is in Settings >> Conditions 
![image](https://github.com/user-attachments/assets/4a8b1cc3-7ce5-4998-b834-2306a6cad6e3)

Using an Update Setting condition like this one below would switch you from Create to Update. You should also add the update setting to switch it back
![image](https://github.com/user-attachments/assets/2a3a6aee-c2ec-46d5-9457-ea03402e978f)
The JavaScript function in the "Is In" setting is:
```JS
let rows = $("New Data Provider.Rows")

let firstNames = rows.map((row) => {
  return row.FirstName
})

return firstNames
```

Which creates an array of the names in the Employees table. Then if we submit the form with no First Name, we create a new row, but if we submit with a name that matches, it writes it to that row.

In this example, for the sake of ease, I set the Email column to be required, and the First Name column to be optional.

Here's an app-export so you can take a closer look at how I've achieved it.
[14172-export-1721061350972.tar.gz](https://github.com/user-attachments/files/16237666/14172-export-1721061350972.tar.gz)

**Comment by Athik321** [comment]
I want to update the row based on my table structure, not the user table. Below is my table structure for driver_blacklist:

CREATE TABLE `driver_blacklist` (
  `id` int UNSIGNED NOT NULL,
  `Date` datetime DEFAULT NULL,
  `license_number` text CHARACTER SET utf8mb4 COLLATE utf8mb4_0900_ai_ci,
  `driver_name` text,
  `driver_number` float(10,0) DEFAULT NULL,
  `adhaar_number` float(15,0) DEFAULT NULL,
  `transpoter_name` text,
  `vehicle_number` text,
  `rc_no` text,
  `blacklist_type` text,
  `reason` text
) 

I have a dropdown field named license_number from which I fetch data from another table through a data provider. When I select a license_number from the dropdown, if the value exists in the driver_blacklist table, upon submitting the form it should update the existing record; otherwise, it should create a new row.
